### PR TITLE
Added additional tests for CA implementations and OIDC

### PR DESCRIPTION
**Pull request by haydentherapper**
Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Review by cpanato — COMMENTED** [review]
lgtm

File: pkg/ca/ephemeralca/ephemeral_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+func TestNewEphemeralCA(t *testing.T) {
+	ca, err := NewEphemeralCA()
+	if err != nil {
+		t.Fatalf("unexpected error generating ephemeral CA: %v", err)
+	}
+	if ca.RootCA == nil {
+		t.Fatalf("ca not set up")
+	}
+	if ca.RootCA.NotAfter.Sub(ca.RootCA.NotBefore) < time.Hour*87660 {
```
Comment (bobcallaway) [inline pkg/ca/ephemeralca/ephemeral_test.go:33]: if you spin this again, doing `time.Hour * 24 * 365 * 10` would be more readable

**Review by bobcallaway — DISMISSED** [review]
lgtm minus one nit

**Review by dlorenc — APPROVED** [review]

**Comment by haydentherapper** [comment]
Review dismissal is far more aggressive now. I thought it had some tolerance for small nits...

File: pkg/ca/ephemeralca/ephemeral_test.go (hunk 1 above)
Comment (haydentherapper) [reply pkg/ca/ephemeralca/ephemeral_test.go]: done

**Comment by dlorenc** [comment]
> Review dismissal is far more aggressive now. I thought it had some tolerance for small nits...

Allstarbot yelled at me so I had to change the setting to dismiss everything :(

**Review by dlorenc — APPROVED** [review]

**Pull request merged**